python/ray/ml/trainer.py

```diff
@@ -54,6 +58,21 @@ def __init__(
 
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def training_function(
+        self, config: Dict[str, Any], checkpoint: Optional[Checkpoint] = None
+    ):
+        """Function to define the training logic.
+
+        `self.train_dataset` and the Dataset values in `self.extra_datasets`
+        have already been preprocessed by `self.preprocessor`.
+
+        Args:
+            config: Configurations for training logic specific implementation.
```

Comment:
Specify where the config will come from? does user provide, etc?

Reply:
To answer your question, this comes from param_space from Tuner.

Which to me, it doesn't really make sense to have Tuner details define the Trainer interface.

Let's discuss this more offline. I made a doc with the different options to consider, please take a look.

Reply:
well, can call this `param_space` then? I think it's a mistake that Tune doesn't do this. I know it all historically probably comes down to running RLlib with Tune and then it doesn't really make sense to call your RLlib configuration `param_space`. But in this case, shouldn't we be explicit? That _is_ what it's here for, no?

Reply:
This arg has been removed.